hive/src/main/java/org/apache/iceberg/hive/legacy/LegacyHiveTableUtils.java

```diff
@@ -0,0 +1,144 @@
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.StructLike;
+import org.apache.iceberg.avro.AvroSchemaUtil;
+import org.apache.iceberg.types.Types;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+class LegacyHiveTableUtils {
+
+  private LegacyHiveTableUtils() {}
+
+  private static final Logger LOG = LoggerFactory.getLogger(LegacyHiveTableUtils.class);
+
+  static Schema getSchema(org.apache.hadoop.hive.metastore.api.Table table) {
+    Map<String, String> props = getTableProperties(table);
+    String schemaStr = props.get("avro.schema.literal");
+    Schema schema;
+    if (schemaStr == null) {
+      LOG.warn("Table {}.{} does not have an avro.schema.literal set; using Hive schema instead. The schema will not" +
+          " have case sensitivity and nullability information", table.getDbName(), table.getTableName());
+      // TODO: Add support for tables without avro.schema.literal
+      throw new UnsupportedOperationException("Reading tables without avro.schema.literal not implemented yet");
+    } else {
+      schema = new Schema(
+          AvroSchemaUtil.convert(new org.apache.avro.Schema.Parser().parse(schemaStr)).asStructType().fields());
+    }
+
+    List<String> partCols = table.getPartitionKeys().stream().map(FieldSchema::getName).collect(Collectors.toList());
+    return addPartitionColumnsIfRequired(schema, partCols);
+  }
+
+  private static Schema addPartitionColumnsIfRequired(Schema schema, List<String> partitionColumns) {
+    List<Types.NestedField> fields = new ArrayList<>(schema.columns());
+    AtomicInteger fieldId = new AtomicInteger(10000);
+    partitionColumns.stream().forEachOrdered(column -> {
+      Types.NestedField field = schema.findField(column);
+      if (field == null) {
+        // TODO: Support partition fields with non-string types
+        fields.add(Types.NestedField.required(fieldId.incrementAndGet(), column, Types.StringType.get()));
+      } else {
+        Preconditions.checkArgument(field.type().equals(Types.StringType.get()),
+            "Tables with non-string partition columns not supported yet");
+      }
+    });
+    return new Schema(fields);
+  }
+
+  static Map<String, String> getTableProperties(org.apache.hadoop.hive.metastore.api.Table table) {
+    Map<String, String> props = new HashMap<>();
+    props.putAll(table.getSd().getSerdeInfo().getParameters());
+    props.putAll(table.getSd().getParameters());
+    props.putAll(table.getParameters());
+    return props;
+  }
+
+  static PartitionSpec getPartitionSpec(org.apache.hadoop.hive.metastore.api.Table table, Schema schema) {
+    PartitionSpec.Builder builder = PartitionSpec.builderFor(schema);
+
+    table.getPartitionKeys().forEach(fieldSchema -> {
+      // TODO: Support partition fields with non-string types
+      Preconditions.checkArgument(fieldSchema.getType().equals("string"),
+          "Tables with non-string partition columns not supported yet");
```

Comment:
Are we aware of any tables that have non-string partition columns?

Reply:
* include the name and the type of the field in the error message.
* "Iceberg only supports STRING type partition columns in Hive tables"
* Remove "yet" unless there is certainty that Iceberg will provide support for all primitive types used as partition columns.

Reply:
Will keep this as is for now since Ratandeep will be sending out a PR addressing this TODO immediately after this